queasars/job_shop_scheduling/random_problem_instances.py

```diff
@@ -0,0 +1,100 @@
+# Quantum Evolving Ansatz Variational Solver (QUEASARS)
+# Copyright 2024 DLR - Deutsches Zentrum für Luft- und Raumfahrt e.V.
+
+from typing import Union, Optional, TypeVar
+from random import Random
+
+from queasars.job_shop_scheduling.problem_instances import Machine, Operation, Job, JobShopSchedulingProblemInstance
+
+
+T = TypeVar("T")
+
+
+def _get_random_value_from_distribution(distribution: dict[T, float], random_generator: Random) -> T:
+    """
+    Returns a random element from a probability distribution with the according likelihood.
+
+    :param distribution: dictionary in which the keys are the elements to be drawn from and the values are the
+        probabilities. The probabilities of all entries must add up to one
+    :type distribution: dict[T, float]
+    :param random_generator: random generator which seeds the randomness
+    :type random_generator: Random
+    :return: a random element from the distribution
+    :rtype: T
+    """
+    if sum(distribution.values()) != 1:
```

Comment:
Hmm, a sum over some floats unequal to 1? This sound dangerous to me. I think python offers `math.isclose()` or similar?

Reply:
Ah, you are right! I adjusted this accordingly.

Reply:
Thanks!